extensions/wikia/ExactTargetUpdates/maintenance/ExactTargetUpdateUserEditsPerWikiMaintenance.php

```diff
@@ -0,0 +1,132 @@
+	 */
+	public function execute() {
+		global $wgDWStatsDB;
+		// Get DB
+		$oStatsDBr = wfGetDB( DB_SLAVE, [], $wgDWStatsDB );
+		$sStartDate = $this->getLastDayDate();
+		$oUsersListResult = $this->getUsersEditedRecently( $oStatsDBr, $sStartDate );
+		$aUsersEditsData = $this->getUserEdits( $oStatsDBr, $sStartDate, $oUsersListResult );
+		$this->addEditsUpdateTask( $aUsersEditsData );
+	}
+
+	private function getUsersEditedRecently( DatabaseBase $oStatsDBr, $sStartDate ) {
+		// Get list of users that made edits in last period
+		$sql = ( new WikiaSQL() )
+			->SELECT()
+			->DISTINCT( 'user_id' )
+			->FROM( 'rollup_wiki_user_events' )
+			->WHERE( 'time_id' )->GREATER_THAN( $sStartDate )
+			->AND_( 'period_id' )->EQUAL_TO( self::DAILY_PERIOD )
+			->AND_( 'user_id' )->NOT_EQUAL_TO( 0 );
+
+		/* @var ResultWrapper $oUsersListResult */
+		$oUsersListResult = $sql->run( $oStatsDBr );
+
+		return $oUsersListResult;
+	}
+
+	/**
+	 * Fetches user edits from statsDB from last period determined by prepareTimeCondition function
+	 * e.g. result
+	 * [ 12345 => [ 177 => 5 ] ]; It means user 12345 made 5 edits on 177 wiki
+	 * @param DatabaseBase $oStatsDBr
+	 * @param string $sStartDate e.g. 2014-12-31
+	 * @param boolean|ResultWrapper $oUsersListResult
+	 * @return array
+	 */
+	private function getUserEdits( DatabaseBase $oStatsDBr, $sStartDate, $oUsersListResult ) {
+		// Get user edits
+		$aUsersEditsData = [];
+		foreach ( $oUsersListResult as $oUserResult ) {
+			if ( !$this->isUserBot( $oUserResult->user_id ) ) {
+				$sql = ( new WikiaSQL() )
+					->SELECT( 'user_id' )
+						->FIELD( 'wiki_id' )
+						->FIELD( 'sum( edits ) + sum( creates )' )->AS_( 'editcount' )
+					->FROM( 'rollup_wiki_user_events' )
+					->WHERE( 'time_id' )->GREATER_THAN( $sStartDate )
+					->AND_( 'period_id' )->EQUAL_TO( self::DAILY_PERIOD )
+					->AND_( 'user_id' )->EQUAL_TO( $oUserResult->user_id )
+					->GROUP_BY( 'wiki_id' );
+
+				/* @var ResultWrapper $oUserEditCountWikisResult */
+				$oUserEditCountWikisResult = $sql->run( $oStatsDBr );
+
+				foreach ( $oUserEditCountWikisResult as $oUserEditCountWikiResult ) {
+					$aUsersEditsData[ $oUserEditCountWikiResult->user_id ][ $oUserEditCountWikiResult->wiki_id ] =
+						intval( $oUserEditCountWikiResult->editcount );
+				}
+			}
+		}
+		return $aUsersEditsData;
+	}
+
+	private function isUserBot($uId) {
+		$aBotsList = $this->getBotsIds();
+		return array_key_exists( $uId, $aBotsList );
+	}
+
+	private function getBotsIds() {
+		if ( $this->aBotsList === NULL ) {
+			$this->loadBotsIds();
+		}
+		return $this->aBotsList;
+	}
+
+	private function loadBotsIds() {
+		global $wgExternalSharedDB;
+		$oExternalSharedDBr = wfGetDB( DB_SLAVE, [], $wgExternalSharedDB );
+		$sql = ( new WikiaSQL() )
+			->SELECT( 'ug_user' )
+			->FROM( 'user_groups' )
+			->WHERE( 'ug_group' )->IN( [ 'bot', 'bot-global' ] );
+
+		/* @var ResultWrapper $oBotsResult */
+		$oBotsResult = $sql->run( $oExternalSharedDBr );
```

Comment:
Since you're going to loop over the result anyway, you could use `$this->aBotsList = $sql->runLoop(...`.